coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LongRunningAction.java

```diff
@@ -319,7 +319,12 @@ public boolean restore_state(InputObjectState os, int ot) {
                     }
 
                     if (status == LRAStatus.Active) {
-                        status = LRAStatus.Cancelling; // transition from Active to Cancelling
+                        if (getSize(pendingList) == 0 && getSize(preparedList) == 0 && getSize(heuristicList) == 0) {
+                            updateState(LRAStatus.Cancelled);
```

Comment:
Please can you share the pre-crash condition for the LRA such that `(getSize(pendingList) == 0 && getSize(preparedList) == 0 && getSize(heuristicList) == 0)` but the status was Active? Is this basically an LRA that had been created but no participants registered so we are just moving it to Cancelled as the TTL is expired. I guess so but if so maybe it's worth a comment because it at least made my curious to know how this could happen.

Reply:
Tha is the case tested here: https://github.com/jbosstm/lra/pull/27/files#diff-1a03090f8ce8ff7635f6bc880a133ff1d5f01a56955be2da86c33199963bf879R151
You are right, it is the case where no participants are registered and the LRA is active. For instance if we create a shortLRA and then the server crashes before the participants are registered, then when the server restarts the LRA is already expired.

Reply:
Thank you, I think it might be worth a comment in the code just to mention that as it's not immediately obvious but I suppose it's not essential to do that. Thank you for the answer on here, though :)

Reply:
@marcosgopen please can I ask you to confirm if this change is related in some way to JBTM-3966 or is it unrelated?

Reply:
And would the second commit be OK in it's own pull request or does it require the first commit? If so I think some linkage between https://issues.redhat.com/browse/JBTM-3966 and https://issues.redhat.com/browse/JBTM-3971 should be made in the JIRAs? I am not sure which but maybe JBTM-3966 should be marked as blocked by JBTM-3971?

Reply:
I think from the JIRA linkage now it can now be understood that JBTM-3971 really should be part of the same pull request (or merged first, so that JBTM-3966 will be more predictable) - thank you!